mod/web_services/lib/client.php

```diff
@@ -158,3 +166,38 @@ function serialise_api_headers(array $headers) {
 
 	return trim($headers_str);
 }
+/**
+ * Encode an array as string to be sended as URLs parameter
+ *
+ * @param array  $arr          The array to be encoded
+ * @param string $perfix       Optional prefix to be used 
+ *
+ * @return string
+ */
+function encode_array($arr, $prefix = null)
+    {
+        if (!is_array($arr)) {
+            return $arr;
+        }
+
+        $r = array();
+        foreach ($arr as $k => $v) {
+            if (is_null($v)) {
+                continue;
+            }
+
+            if ($prefix && $k && !is_int($k)) {
```

Comment:
if ($prefix && isset($k)) {

should solve the issue when using "0" as array key